advanced/matrix/matrix.py

```diff
@@ -0,0 +1,329 @@
+class Matrix:
+    def __init__(self, *args):
+        """
+        Initialize method - Organize the Matrix checks the input as well.
+        """
+        # Remove unnecessary brackets. Like so: (((1, 2), (3, 4)),)
+        if type(args[0][0]) == tuple:
+            self._matrix = args[0]
+        else:
+            self._matrix = args
+
+        self.tuples = self._matrix
```

Comment:
You don't want the user to get your original tuples.
read about `@property`, it a way to write a getter function to a property